frontend/src/components/interaction-manager/content/log.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from 'react'
+
+import { Intent } from '../../../config/intent'
+import { AppToaster } from '../../toaster'
+import type { InteractionContentProps, Log } from '../types'
+
+const getMessageIntent = (type: string) => {
+  switch (type) {
+    case 'error': {
+      return Intent.DANGER
+    }
+    case 'warning': {
+      return Intent.WARNING
+    }
+    case 'success': {
+      return Intent.SUCCESS
+    }
+    default: {
+      return Intent.NONE
+    }
+  }
+}
+
+export const LogComponent = ({
+  interaction,
+  isResolved,
+  setResolved
+}: InteractionContentProps<Log>) => {
+  useEffect(() => {
```

Comment:
question (blocking): As this only does hook things why does it need to be a full component? Can we not just write this as a hook?

Reply:
Same with other files of this type returning null

Reply:
Agree with @dexturr

Reply:
some of the problems:
- i can't use hooks conditionally, i couldn't just make this a `useLog()` hook and call it in a switch statement, or whatever method i'd use to differentiate between which event i need to handle
- i could just potentially have one hook, which would need to return a component to render, but that feel just so wrong for many reasons:
    - it's more difficult to have a control over the component props
    - would hide the hierarchy of how things render
    - hooks are usually to extract component logic from components, not to render them (in case of the ones which need to actually render content)
    - etc. 

This component renders a notification, I could potentially return the notification element, but that would also require some refactor, as the AppToaster has been designed to have a central place for notifications, hence the weird show call instead of returning jsx.